packages/ui5-proxy-middleware/src/base/utils.ts

```diff
@@ -195,36 +195,40 @@ export const setHtmlResponse = (res: any, html: string): void => {
  * @param args list of runtime arguments
  * @returns The content of the manifest.json
  */
-export const getManifest = async (args: string[]): Promise<Manifest> => {
+export async function getManifest(args: string[]): Promise<Manifest | undefined> {
     const projectRoot = process.cwd();
     const yamlFileName = getYamlFile(args);
     const ui5YamlPath = join(projectRoot, yamlFileName);
     const webAppFolder = await getWebAppFolderFromYaml(ui5YamlPath);
     const manifestPath = join(projectRoot, webAppFolder, 'manifest.json');
-    const manifest: Manifest = JSON.parse(readFileSync(manifestPath, { encoding: 'utf8' }));
-
-    return manifest;
-};
+    if (existsSync(manifestPath)) {
+        return JSON.parse(readFileSync(manifestPath, { encoding: 'utf8' }));
+    } else {
+        return undefined;
+    }
+}
 
 /**
  * Gets the minUI5Version from the manifest.json.
  *
  * @param args list of runtime args
+ * @param manifest optional already loaded manifest.json
  * @returns The minUI5Version from manifest.json or undefined otherwise
  */
-export async function getUI5VersionFromManifest(args: string[]): Promise<string | undefined> {
-    const manifest = await getManifest(args);
-    return manifest['sap.ui5']?.dependencies?.minUI5Version;
+export async function getUI5VersionFromManifest(args: string[], manifest?: Manifest): Promise<string | undefined> {
+    const mf = manifest ?? (await getManifest(args));
+    return mf?.['sap.ui5']?.dependencies?.minUI5Version;
 }
 
 /**
  * Determines which UI5 version to use when previewing the application.
  *
  * @param version ui5 version as defined in the yaml or via cli argument
  * @param log logger for outputing information from where ui5 version config is coming
+ * @param manifest optional already loaded manifest.json
  * @returns The UI5 version with which the application will be started
  */
-export async function resolveUI5Version(version?: string, log?: ToolsLogger): Promise<string> {
+export async function resolveUI5Version(version?: string, log?: ToolsLogger, manifest?: Manifest): Promise<string> {
```

Comment:
I propose to make `manifest.json` mandatory. Function `resolveUI5Version` is not in the public interface and used in one place in `middleware.ts` only. This would make two other functions obsolete and we can remove them:
1. `getUI5VersionFromManifest()`
2. `getManifest()` -> this is especially good as it contains the problematic `process.cwd()` statement and with removing this we minimize the risk a similar issue occurs again

Whenever we need the manifest again, we pass it from `middleware.ts`.

What do you think?

Reply:
Agree, done with https://github.com/SAP/open-ux-tools/pull/1054/commits/e66599a445bf869d90bb045cd8e762f64fbb6128